lib/multi.c

```diff
@@ -1156,10 +1215,24 @@ static CURLMcode Curl_multi_wait(struct Curl_multi *multi,
 
       data = data->next; /* check next handle */
     }
+
+#ifdef USE_WINSOCK
+    ++nevents;
+#endif
   }
 
   /* Add external file descriptions from poll-like struct curl_waitfd */
   for(i = 0; i < extra_nfds; i++) {
+#ifdef USE_WINSOCK
+    long events = 0;
+    if(extra_fds[i].events & CURL_WAIT_POLLIN)
+      events |= FD_READ;
+    if(extra_fds[i].events & CURL_WAIT_POLLPRI)
+      events |= FD_OOB;
+    if(extra_fds[i].events & CURL_WAIT_POLLOUT)
+      events |= FD_WRITE;
+    WSAEventSelect(extra_fds[i].fd, uevents[nevents], events);
```

Comment:
Please perform error checking like it is done here:
https://github.com/curl/curl/blob/f9983a6f9eb5314ad1a788d06929f6763ada6204/tests/server/sockfilt.c#L744-L745